core/src/rpc.rs

```diff
@@ -625,7 +625,7 @@ impl JsonRpcRequestProcessor {
             self.check_slot_cleaned_up(&result, slot)?;
             Ok(result.ok())
         } else {
-            Ok(None)
+            Err(RpcCustomError::BlockNotAvailable { slot }.into())
```

Comment:
We actually could break out these rpc error-handling changes and get them landed sooner.  I guess I could have done that in June in retrospect!

Reply:
Ah, good call. I'll pull that out to land & backport first.